sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -171,6 +171,9 @@ case class DropFunctionCommand(
       if (FunctionRegistry.builtin.functionExists(FunctionIdentifier(functionName))) {
         throw QueryCompilationErrors.cannotDropNativeFuncError(functionName)
       }
+      if (catalog.isCustomBuiltinFunction(FunctionIdentifier(functionName))) {
+        throw QueryCompilationErrors.cannotDropCustomBuiltInFuncError(functionName)
+      }
```

Comment:
dropping a custom built-in function is fine. but it's weird if we can drop a built-in function.